homeassistant/components/cert_expiry/config_flow.py

```diff
@@ -36,18 +39,37 @@ def _prt_in_configuration_exists(self, user_input) -> bool:
             return True
         return False
 
+    def _test_connection(self, user_input=None):
+        """Test connection to the server and try to get the certtificate."""
+        try:
+            address = (user_input[CONF_HOST], user_input.get(CONF_PORT, DEFAULT_PORT))
+            with socket.create_connection(address, timeout=TIMEOUT) as sock:
```

Comment:
Can we break out this code into a helper function that is used both here and in the entity?